pkgs/build-support/node/fetch-yarn-deps/default.nix

```diff
@@ -174,6 +210,28 @@ in
     };
   } ./yarn-config-hook.sh;
 
+  yarnBerry3ConfigHook = makeSetupHook {
```

Comment:
Personally I like how in `pnpm` they put all their hooks under the package itself (e.g. `pnpm_9.configHook`).

This way you can just:
```nix
let
  yarn = yarn-berry_3;
in
{
  yarnOfflineCache = yarn.fetchDeps { ... };
  nativeBuildInputs = [ yarn yarn.configHook ];
}
```

Reply:
> With your suggestion, it will mean a refactor of all packages using `yarn` to now use `yarn.fetchYarnDeps`.

Small correction: `yarn.fetchDeps` :).

> This can of course be done, but IMHO would be too much for this PR. Lets keep it in mind and maybe do this refactor in a separate PR or tracking issue. This is what @doronbehar suggested for the long-term ecosystem for the `src` argument ([#355053 (comment)](https://github.com/NixOS/nixpkgs/pull/355053#discussion_r1847179151))

Furthermore, this suggestion can ease the refactoring: We could make `yarn.fetchDeps` get the new signature with `src`, and simply make `fetchYarnDeps` throw a message like `fetchYarnDeps is deprecated in favor of yarn.fetchDeps which accepts an src argument instead of yarnLock argument`. This would allow us to not make the effort of making the same function compatible with both signatures.

Reply:
I mean we could easily just make these as aliases. But agree that this couls be done in another PR.

Reply:
Ok, lets target that for another PR